tests/test_calendar_analyzer.py

```
@@ -138,19 +156,34 @@
     monkeypatch.setattr("sys.argv", [
         "calendar_analyzer.py",
-        "--calendar", "/tmp/dummy.ics",
+        "--calendar", dummy_path,
         "--start-date", "2024-02-29"  # Valid leap year date
     ])
 
     with pytest.raises(SystemExit):  # Will fail because dummy file doesn't exist
         calendar_analyzer.main()
 
     # Test invalid leap year date
+    dummy_file2 = tempfile.NamedTemporaryFile(suffix=".ics", delete=False)
```

Ruff (0.11.9)

Use a context manager for opening files

(SIM115)